cql3/statements/create_table_statement.cc

```diff
@@ -434,6 +436,23 @@ std::optional<sstring> check_restricted_table_properties(
     // function before cfprops.validate() (there, validate() is only called
     // in prepare_schema_mutations(), in the middle of execute).
     auto strategy = cfprops.get_compaction_strategy_class();
+    sstables::compaction_strategy_type current_strategy;
+    gc_clock::duration current_ttl;
+    // cfprops doesn't return any of the table attributes unless the attribute
+    // has been specified in the CQL statement. Validate whether the table
+    // already exists. If it does, this was an ALTER TABLE statement.
+    try {
+        schema_ptr schema = validation::validate_column_family(qp.db(), keyspace, table);
+        current_strategy = schema->compaction_strategy();
+        current_ttl = schema->default_time_to_live();
+    } catch (exceptions::invalid_request_exception) {
+        current_strategy = sstables::compaction_strategy_type::null;
+        current_ttl = gc_clock::duration::zero();
+    }
+
```

Comment:
Better to add an optional schema (std::optional<schema>) in check_restricted_table_properties() parameter list. So we won't have to embed this knowledge in check_restricted_table_properties(). So alter table calls it with schema, whereas create table calls it with std::nullopt.

Reply:
I agree. That's what happens when an Architect pretends to add things. I will change it once I get a consensus on the failing dtest thingy, there's no point on pushing changes when these are going to fail anyway on CI.

Reply:
Don't worry. You did a great job with this series. This protection is very important for the product.